src/NuGet.Core/NuGet.DependencyResolver/SourceRepositoryDependencyProvider.cs

```diff
@@ -39,9 +62,21 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
         {
             await EnsureResource();
 
-            var packageVersions = await _findPackagesByIdResource.GetAllVersionsAsync(libraryRange.Name, cancellationToken);
+            IEnumerable<NuGetVersion> packageVersions = null;
+            try
+            {
+                packageVersions = await _findPackagesByIdResource.GetAllVersionsAsync(libraryRange.Name, cancellationToken);
+            }
+            catch (FatalProtocolException e) when (_ignoreFailedSources)
+            {
+                if (!_ignoreWarning)
+                {
+                    _logger.LogWarning(e.Message);
```

Comment:
I don't think this should log here since in some cases it logs at a lower level and there will be duplicate warnings.